### PR TITLE
Add apiary documentation

### PR DESCRIPTION
**Pull request by jackrootstrap**
#### Trello board reference:

* [Trello Card #32 Apiary-documentation](https://trello.com/c/5EaTmjIA/32-apiary-documentation)

---

#### Description:
* Add apiary documentation

#### Notes:

* Updated gemfile
* Overrided sessions controller
* Updated registration controller
* Updated locals
* Generated apiary documentation
* Created helper methods for testing

### CODE REVIEW

File: doc/api/index.html [hunk 1]
```diff
@@ -0,0 +1,137 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>API Documentation</title>
+  <meta charset="utf-8">
+  <style>
+    
+body {
```
Comment (JulianPasquale) [inline doc/api/index.html:8]: Can we move the CSS code to a different file? Same suggestion for the rest of the html files

File: config/locales/en.yml [hunk 2]
```diff
@@ -33,3 +33,7 @@ en:
   api:
     errors:
       invalid_body: Please submit proper sign up data in request body.
+      confirmation_email: A confirmation email was sent to your account
```
Comment (JulianPasquale) [inline config/locales/en.yml:36]: This doesn't look like an error, maybe this is not the best place for it?

**Review by JulianPasquale — COMMENTED** [review]
Please take a look at our [tech_guides](https://github.com/rootstrap/tech-guides/tree/master/pull-request#opening-a-pull-requestpr) and the suggested PR size 🙂

File: doc/api/index.html (hunk 1 above)
Comment (jackrootstrap) [reply doc/api/index.html:8]: This files are auto-generated. I think these files should not be updated, wdyt?

File: config/locales/en.yml (hunk 2 above)
Comment (jackrootstrap) [reply config/locales/en.yml:36]: This message actually comes when a user tries to login when is not confirmed yet. This message also appears in the response error key.

File: spec/acceptance/sessions/sign_in_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
```
Comment (sebastiancaraballo) [inline spec/acceptance/sessions/sign_in_spec.rb:2]: We should stick to request tests. @JulianPasquale wdyt?

**Review by sebastiancaraballo — COMMENTED** [review]
Documentation should go on a different PR

**Comment by jackrootstrap** [comment]
@JulianPasquale I already clean the PR to reduce the PR size, but all these files are required here.

**Comment by jackrootstrap** [comment]
@sebastiancaraballo This PR will be handling the apiary documentation!

File: spec/acceptance/sessions/sign_in_spec.rb (hunk 3 above)
Comment (JulianPasquale) [reply spec/acceptance/sessions/sign_in_spec.rb:2]: Yeah, agree

**Review by sebastiancaraballo — APPROVED** [review]

File: spec/requests/registrations_spec.rb [hunk 4]
```diff
@@ -1,56 +0,0 @@
-# frozen_string_literal: true
```
Comment (JulianPasquale) [inline spec/requests/registrations_spec.rb:1]: I think we should not delete this file, but restore the master version of it so it stays unchanged

Comment (jackrootstrap) [reply spec/requests/registrations_spec.rb:1]: This changes make this spec fails, so I just removed it. This spec will be handled in this [PR](https://github.com/jackrootstrap/rs-target/pull/7)

File: spec/acceptance/sessions/sign_in_spec.rb (hunk 3 above)
Comment (JulianPasquale) [reply spec/acceptance/sessions/sign_in_spec.rb:2]: @jackrootstrap Can you address this comment, please?

Comment (jackrootstrap) [reply spec/acceptance/sessions/sign_in_spec.rb:2]: This was already addressed with @JulianPasquale offline!

**Review by JulianPasquale — APPROVED** [review]

**Pull request merged**